src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -589,5 +600,21 @@ public void onPatchGamePrefsForMultipleInstances() {
       notificationService.addImmediateErrorNotification(e, "settings.fa.patchGamePrefsFailed");
     }
   }
+
+  public void onAddMirrorURL() {
+    URI uri = URI.create(mirrorURITextField.getText());
+    if (mirrorURITextField.getText().isEmpty() || mirrorURLsListView.getItems().contains(uri)) {
+      return;
+    }
+    try {
+      uri.toURL();
+    } catch (Exception e) {
```

Comment:
```suggestion
    } catch (MalformedURLException e) {
```

Reply:
No, I had that and then it threw an IllegalArgumentException

Reply:
Then this should onlt check for MalformedURLException and IllegalArgumentException

Also the try should really be around the whole method body since URI.create can also throw the IllegalArgumentException if it doesn't match the spec. If you want to check both the URL and URI at the same time I would just do
`URI uri = new URL(text).toURI()`
and add the try catch around that